tests/acceptance/17_packages/10_new/unsafe/the_great_package_test_generator.py

```diff
@@ -0,0 +1,542 @@
+
+        promises.append(promise)
+
+    return promises
+
+
+# Given the transitions between the two version, determine if the change of one
+# architecture package will trigger a change in the other. Only one of the two
+# inputs can be different, but the returned value may cause both to be
+# different. This is only relevant for "repo" style promises, since "file"
+# promises will never touch more than one package at a time.
+def resolve_arch_conflicts(from_64, from_32, to_64, to_32):
+    if from_64 != to_64:
+        if from_32 != "0":
+            if to_64 >= from_32:
+                to_32 = to_64
+            else:
+                to_32 = "0"
+
+    elif from_32 != to_32:
+        if from_64 != "0":
+            if to_32 >= from_64:
+                to_64 = to_32
+            else:
+                to_64 = "0"
+
+    return to_64, to_32
+
+
+# Simulate, with the given promise, what the state of the system would be,
+# if we started from the versions in from_64 and from_32.
+def simulate_promise(promise, from_64, from_32):
+    to_64 = from_64
+    to_32 = from_32
+
+    arch, version = promise.get('arch'), promise.get('version')
+    if promise["policy"] == "present":
+        if promise["type"] == "repo":
+            if version == "latest":
+                version = "2"
+
+            if arch == "64_bit":
+                if version:
+                    to_64 = version
+                elif from_64 == "0":
+                    to_64 = "2"
+
+                to_64, to_32 = resolve_arch_conflicts(from_64, from_32, to_64, to_32)
+
+            elif arch == "32_bit":
+                if version:
+                    to_32 = version
+                elif from_32 == "0":
+                    to_32 = "2"
+
+                to_64, to_32 = resolve_arch_conflicts(from_64, from_32, to_64, to_32)
+
+            elif version:
+
+                if from_64 != "0" or (from_64 == "0" and from_32 == "0"):
+                    to_64 = version
+
+                if from_32 != "0":
+                    to_32 = version
+
+            elif from_64 == "0" and from_32 == "0":
+                to_64 = "2"
+
+        else:
+            # Todo: Errors when file doesn't match arch/version?
+            if promise["file_arch"] == "64_bit":
+                to_64 = promise["file_version"]
+            else:
+                to_32 = promise["file_version"]
+
+    else:
+        if arch == "64_bit":
+            if not version or version == from_64:
+                to_64 = "0"
+        elif arch == "32_bit":
+            if not version or version == from_32:
+                to_32 = "0"
+        else:
+            if version:
+                if version == from_64:
+                    to_64 = "0"
+                if version == from_32:
+                    to_32 = "0"
+            else:
+                to_64 = "0"
+                to_32 = "0"
+
+    return to_64, to_32
+
+
+# Calculate all possible transitions from from_state, to to_state.
```

Comment:
Doubling 'from' and 'to' is ugly.  Might be clearer as
`# Calculate all possible transitions from_state -> to_state`